packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/EmailPasswordAuthTests.kt

```diff
@@ -152,45 +153,50 @@ class EmailPasswordAuthWithAutoConfirmTests {
         Unit
     }
 
-    // @Test
-    // fun callResetPasswordFunction() {
-    //    val provider = app.emailPasswordAuth
-    //    val adminApi = app.asTestApp
-    //    runBlocking {
-    //        adminApi.setResetFunction(enabled = true)
-    //        val email = TestHelper.getRandomEmail()
-    //        provider.registerUser(email, "123456")
-    //        try {
-    //            provider.callResetPasswordFunction(email, "new-password", "say-the-magic-word", 42)
-    //            val user = app.login(Credentials.emailPassword(email, "new-password"))
-    //            user.logOut()
-    //        } finally {
-    //            adminApi.setResetFunction(enabled = false)
-    //        }
-    //    }
-    // }
-    //
-    // @Test
-    // fun callResetPasswordFunction_invalidServerArgsThrows() {
-    //    val provider = app.emailPassword
-    //    admin.setResetFunction(enabled = true)
-    //    val email = TestHelper.getRandomEmail()
-    //    provider.registerUser(email, "123456")
-    //    try {
-    //        provider.callResetPasswordFunction(email, "new-password", "wrong-magic-word")
-    //    } catch (error: AppException) {
-    //        assertEquals(ErrorCode.SERVICE_UNKNOWN, error.errorCode)
-    //    } finally {
-    //        admin.setResetFunction(enabled = false)
-    //    }
-    // }
-    //
-    // @Test
-    // fun callResetPasswordFunction_invalidArgumentsThrows() {
-    //    val provider = app.emailPassword
-    //    assertFailsWith<IllegalArgumentException> { provider.callResetPasswordFunction(TestHelper.getNull(), "password") }
-    //    assertFailsWith<IllegalArgumentException> { provider.callResetPasswordFunction("foo@bar.baz", TestHelper.getNull()) }
-    // }
+    @Test
+    fun callResetPasswordFunction() {
+        val provider = app.emailPasswordAuth
+        val adminApi = app.asTestApp
+        runBlocking {
+            adminApi.setResetFunction(enabled = true)
+            val email = TestHelper.randomEmail()
+            provider.registerUser(email, "123456")
+            try {
+                provider.callResetPasswordFunction(email, "new-password", "say-the-magic-word", 42)
+                val user = app.login(Credentials.emailPassword(email, "new-password"))
+                user.logOut()
+            } finally {
+                adminApi.setResetFunction(enabled = false)
+            }
+        }
+    }
+
+    @Test
+    fun callResetPasswordFunction_invalidServerArgsThrows() {
+        val provider = app.emailPasswordAuth
+        val adminApi = app.asTestApp
+        runBlocking {
```

Comment:
I just modified the old tests slightly, mainly also wrapping them in `runBlocking` to satisfy coroutine scope requirements, but not sure if more test cases are necessary or if there is anything I should change.